### PR TITLE
Subheader generation performance improved

### PR DESCRIPTION
**Pull request by vkjr**
fixes # TBD

### Summary

Length calculation in message preview was incorrect and as a result huge preview texts could be generated from huge messages. That wasn't visible in UI due to visual text truncation but affected performance on Android. Now that fixed and short preview components generated.

#### Platforms
- Android
- iOS

#### Areas that maybe impacted
Last message preview in a chat list

##### Functional

- 1-1 chats
- public chats
- group chats

##### Non-functional
- CPU performance / speed of the app

### Steps to test
- Open Status
- Open chat and post huge message
- Get back to chats preview
- Make sure there is no performance penalty when opening chat 

status: ready

### CODE REVIEW

File: src/status_im/ui/screens/home/views/inner_item.cljs [hunk 1]
```diff
@@ -13,60 +13,47 @@
...
 
-    "blockquote"
-    (conj acc (.substring literal 0 (dec (.-length literal))))
+                 "status-tag"
+                 (str "#" literal)
 
-    "codeblock"
-    (conj acc (.substring literal 0 (dec (.-length literal))))
+                 "link"
+                 destination
```
Comment (0kok0) [inline src/status_im/ui/screens/home/views/inner_item.cljs]: How does a truncated link look, can it be opened from preview?

Comment (vkjr) [reply src/status_im/ui/screens/home/views/inner_item.cljs]: It looks like this. No, it can't be opened from preview

Screenshot 2020-08-17 at 12 07 41

**Review by cammellos — CHANGES_REQUESTED** [review]
I don't think using `generate-gfy` will cut it, it means that we won't show any ENS username for example.
A subscription should be used instead in this cases, is that causing the issue?

**Comment by vkjr** [comment]
@cammellos, I tried to recursively construct nested components and calculate their length but that looked quite clumsy and calculation should rerun whenever we have update on ens name. Let me update PR and try it again to save correct names.

**Comment by cammellos** [comment]
@vkjr what happens if you just keep the code as it is now, and restore `mention-element` as it was before?

File: src/status_im/ui/screens/home/views/inner_item.cljs [hunk 2]
```diff
@@ -13,60 +13,47 @@
...
 
-    "mention"
-    (conj acc [react/text-class
-               [mention-element literal]])
```
Comment (cammellos) [inline src/status_im/ui/screens/home/views/inner_item.cljs:39]: Here, if you just use this instead, and uncomment mention-element, what is the behavior?

**Comment by qoqobolo** [comment]
With build [60c247c](https://github.com/status-im/status-react/commit/60c247caa128bf505c6912710b63e9205e0b80b4) next scenarios have been tested :

1. Changes in time of preview generation compared to the Release build. Tested on the case when several chats (all types) are in Home screen and contain huge messages (>10000 characters);
2. Messages with text formatting;
3. Login to multiaсcount containing several chats with huge unread messages;
4. Navigate to Home screen from other tabs;
5. Put app to background and back;
6. [This](https://docs.google.com/document/d/1XueoOwwBN9cgbahpFJmgtetTKAN7gMkmQW828-qX7GI/edit?usp=sharing) text
7. Preview of chats containing stickers, audio messages, and commands.

**Review by cammellos — APPROVED** [review]

**Comment by vkjr** [comment]
@qoqobolo, pr is ready for testing subheader performance... again :)

**Pull request merged**